src/serac/physics/utilities/finite_element_vector.hpp

```diff
@@ -0,0 +1,218 @@
+// Copyright (c) 2019-2021, Lawrence Livermore National Security, LLC and
+// other Serac Project Developers. See the top-level LICENSE file for
+// details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+/**
+ * @file finite_element_vector.hpp
+ *
+ * @brief This file contains the declaration of structure that manages vectors
+ * derived from an MFEM finite element space
+ */
+
+#pragma once
+
+#include "mfem.hpp"
+
+#include "serac/physics/utilities/variant.hpp"
+
+namespace serac {
+
+/**
+ * @brief Class for encapsulating the data associated with a vector derived
+ * from a MFEM finite element space. Specifically, it contains the information
+ * needed for both primal finite element state fields and dual finite element vectors.
+ *
+ * Namely: Mesh, FiniteElementCollection, FiniteElementVector,
+ * GridFunction, and a distributed vector of the solution
+ */
+class FiniteElementVector {
+public:
+  /**
+   * @brief Structure for optionally configuring a FiniteElementVector
+   */
+  // The optionals are explicitly default-constructed to allow the user to partially aggregrate-initialized
+  // with only the options they care about
```

Comment:
"optionals" seems like a typo

this might be good to include in the doxygen comment too, so it's apparent that the user can use aggregate initialization and only specify which values they want to override